public/js/helper.js

```diff
@@ -1,45 +1,22 @@
-function clone(obj) {
-	return JSON.parse(JSON.stringify(obj));
-}
+const clone = obj => JSON.parse(JSON.stringify(obj));
 
-function isEmpty(obj) {
-    for(var key in obj) {
-        if(obj.hasOwnProperty(key))
-            return false;
-    }
-    return true;
-}
+// todo: I think Object.keys(obj).some(_=>_) should work as well.
+const isEmpty = obj => Object.keys(obj).some(key => obj.hasOwnProperty(key));
```

Comment:
I think the logic is reversed here

Reply:
Indeed. Nice catch!

Reply:
By the way, do you confirm that we can remove hasOwnProperty?

[Try it online!](https://tio.run/##y0osSyxOLsosKNHNy09J/f8/OT@vuEQhs9g1t6CkUsFWIT8pS8HWTkHRPykrNblELzu1slgDKKapV5yfm6oB5IJkgQJ6GYnF/uV5AUX5BalFJZUgGU1Nay4U44wImhcPkosH6gNrzM9J1cvJT9eAateortXUxC6RaGWIQ84Ipy4jmLb//wE "JavaScript (Node.js) – Try It Online")

Reply:
Actually if you're using Object.keys, you can just test Object.key(obj).length === 0; But it's slower than the original method, I'd rather keep it :)